tests/tokens_service_test.go

```diff
@@ -0,0 +1,101 @@
+package tests
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/google/uuid"
+
+	"github.com/flow-hydraulics/flow-wallet-api/jobs"
+	"github.com/flow-hydraulics/flow-wallet-api/tests/internal/test"
+	"github.com/flow-hydraulics/flow-wallet-api/transactions"
+)
+
+func Test_TokensSetup(t *testing.T) {
+	cfg := test.LoadConfig(t, testConfigPath)
+	svc := test.GetServices(t, cfg).GetTokens()
+
+	_, testAccount, err := test.GetServices(t, cfg).GetAccounts().Create(context.Background(), true)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	type input struct {
+		sync      bool
+		tokenName string
+		address   string
+	}
+	type expect struct {
+		job   *jobs.Job
+		tx    *transactions.Transaction
+		error bool
+	}
+	testCases := []struct {
+		name   string
+		input  input
+		expect expect
+	}{
+		{
+			name: "success",
+			input: input{
+				sync:      false,
+				tokenName: "flowToken",
+				address:   testAccount.Address,
+			},
+			expect: expect{
+				job: &jobs.Job{
+					Type:                   transactions.TransactionJobType,
+					State:                  jobs.Init,
```

Comment:
This should be `jobs.Complete` once the test case waits for the job to execute (see comment on L82)